LRP: Fix node-local redirection exceptions (#26144)
cilium-team/cilium: Documentation/operations/upgrade.rst

* Cilium Envoy DaemonSet is now enabled by default, and existing in-container installs
  will be changed to DaemonSet mode unless specifically opted out of. This can be done by
  disabling it manually by setting ``envoy.enabled=false`` accordingly.

> * Local Redirect Policy when enabled with socket-based load-balancing allows traffic going
>   from policy selected node-local backends destined to the policy frontend, and redirected back
>   to the backends, by default. In order to skip redirection from backends, create policies
>   with the ``skipRedirectFromBackend`` flag set to ``false``.

Reviewer: Wording suggestion for clarity:

```suggestion
* Local Redirect Policy, when enabled with socket-based load-balancing, redirects traffic
  from policy-selected node-local backends destined to the policy's frontend, back to the
  backends listed in the policy. To override this behavior, which is enabled by default, create
  local redirect policies with the ``skipRedirectFromBackend`` flag set to ``false``.
```
